autoread.js

```diff
@@ -101,4 +104,23 @@ function addSpeakBox(){
 	window.addEventListener('touchstart', playHotKey, true);
 
 	sbPlay.focus();
+}
+
+function handleCheckbox(){
+	let inputElems = currentLine.getElementsByTagName("input")
+	if(inputElems.length > 0 && inputElems[0].type == "checkbox"){
+		inputElems[0].click();
+	}
+}
+
+function handleSublist(){
+	let lineToRead;
+	//if we run into a sublist, only read the first line of the sublist.
+	if(currentLine.getElementsByTagName("ol").length > 0 || currentLine.getElementsByTagName("ul").length > 0){
+		lineToRead = currentLine.innerText.split('\n')[0];
```

Comment:
Have you considered using currentLine.childNodes instead of parsing the inner text directly?

Reply:
we still have to parse out the non sub list textContent from the array of nodes. 

But we could do it this way, this would also let us parse out more elements if needed in the future.

If we want to go down this write, I may/can rewrite the checkbox handling to just be parsed here too since we're already parsing for element types here.

thoughts?
```
function handleSublist(currentLine){
	let lineToRead;
	//if we run into a sublist, only read the first line of the sublist.
	if(currentLine.getElementsByTagName("ol").length > 0 || currentLine.getElementsByTagName("ul").length > 0){
		//lineToRead = currentLine.innerText.split('\n')[0];
		let str = "";
		for (let i = 0; i < currentLine.childNodes.length; i++) {
			let elem = currentLine.childNodes[i];
			if(!(elem.nodeName == "OL" || elem.nodeName == "UL")){
				str += elem.textContent;
			}
		}
		lineToRead = str;
	}
	else {
		lineToRead = currentLine.innerText;
	}
	return lineToRead;
}
```

Reply:
Yeah, looks good. If we're checking for nodeName in the inner line, we could get rid of the outer if statement as it is now redundant.
You could further combine it all into a single line like
```
Array.of(...currentLine.childNodes)
.filter(x=>x.nodeName != "OL" && x.nodeName != "UL")
.map(x=>x.textContent)
.join('')
```

Reply:
alright, I implemented this in the latest commit 
8a2fc18